server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -158,8 +157,29 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
     }
 
     @Override
-    public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) throws IOException {
-        IOUtils.rm(blobNames.stream().map(path::resolve).toArray(Path[]::new));
+    public void deleteBlobsIgnoringIfNotExists(Iterator<String> blobNames) throws IOException {
+        IOException ioe = null;
+        long suppressedExceptions = 0;
+        while (blobNames.hasNext()) {
+            try {
+                IOUtils.rm(path.resolve(blobNames.next()));
+            } catch (IOException e) {
+                // track up to 10 delete exceptions and try to continue deleting on exceptions
+                if (ioe == null) {
+                    ioe = e;
+                } else if (ioe.getSuppressed().length < 10) {
+                    ioe.addSuppressed(e);
+                } else {
+                    ++suppressedExceptions;
+                }
+            }
+        }
+        if (ioe != null) {
+            if (suppressedExceptions > 0) {
+                throw new IOException("Failed to delete files, suppressed [" + suppressedExceptions + "] failures");
```

Comment:
This loses `ioe` altogether? Maybe this:

```suggestion
                ioe.addSuppressed(new IOException("Failed to delete files, suppressed [" + suppressedExceptions + "] failures"));
```